src/plone/api/content.py

```diff
@@ -130,7 +131,10 @@ def get(path=None, UID=None):
             )
 
         try:
-            return site.restrictedTraverse(path)
+            content = site.restrictedTraverse(path)
+            # Only return a content if it implements DynamicType,
+            # which is true for Dexterity content and Comment (plone.app.discussion)
+            return content if isinstance(content, DynamicType) else None
```

Comment:
Generally it is better to keep the `try/except` as small as possible.

So something like this:
```python
try:
    content = site.restrictedTraverse(path)
except (KeyError, AttributeError):
    return None
if $CONDITION:
    return content
```

About `$CONDITION` I am pretty surprised by your choice of checking if content is an instance of the `DynamicType` mixin class.

Is there a reason for that?

Maybe given that we just don't want to return by mistake some object attribute, what about having the condition be:
```python:
from persistent import Persistent
if isinstance(content, Persistent):
    return content
```